hilti/runtime/src/fiber.cc

```diff
@@ -288,11 +573,21 @@ void detail::yield() {
     context::detail::get()->resumable = r;
 }
 
+void detail::checkStack() {
+    auto* fiber = context::detail::get()->fiber.current;
+
+    if ( fiber->type() == Fiber::Type::Main )
+        return;
+
+    if ( fiber->stackBuffer().liveRemainingSize() < configuration::get().fiber_min_stack_size )
+        throw StackSizeExceeded("allowed stack space exceeded");
```

Comment:
Since this is user-visible we could give them something closer to what they were trying to do, e.g.,

```suggestion
        throw StackSizeExceeded("not enough stack space remaining");
```

or even (might be hard for testing)

```suggestion
        throw StackSizeExceeded(fmt("allowed stack space exceeded (%d% consumed)", static_cast<int>(100 *  fiber->stackBuffer().liveRemainingSize() / configuration::get().fiber_min_stack_size)));
```

Reply:
I'll go with the former as the latter would always be reflecting essentially just the limits that are configured. I am planing to add some bookkeeping for stack usage, so that we generally get an idea how large these stacks becomes.